src/error.rs

```diff
@@ -1,11 +1,19 @@
+
 use defines::AfError;
 use std::error::Error;
+use std::sync::Mutex;
 
 pub type ErrorCallback = Fn(AfError);
 
 pub static mut HANDLE_ERROR: &'static ErrorCallback = &handle_error_general; 
```

Comment:
I think i tried reference, but that doesn't work because it will require life time specifier. Adding 'static life time brings us back to the old problem of not being able to create a Mutex around 'static life time object.

I don't remember trying Box but i suspect it will have similar problem. But will give it a shot.